src/main/java/eu/eutampieri/catacombs/model/Boss.java

```diff
@@ -60,15 +68,30 @@ public Boss(final int x, final int y, final TileMap tileMap) {
                 changeDirection();
             }
         }
-
+        if (others.stream().filter((x) -> x instanceof Player)
+                .findFirst()
+                .get()
+                .getHitBox()
+                .overlaps(this.radarBox) && this.weapon.canFire) {
+            setShootingDirection(others.stream().filter((x) -> x instanceof Player).findFirst().get());
+        } else {
+            this.weapon.setCanFire(false);
+        }
         super.update(delta, others);
         updateRadarBoxLocation();
+        weapon.update(delta, others);
+        if (this.weapon.canFire && this.getShootingDirection().getX() != 0 && this.getShootingDirection().getY() != 0) {
+            return weapon.fire((int)getShootingDirection().getX() * weapon.ps, (int)getShootingDirection().getY() * weapon.ps);
+        }
         return List.of();
     }
 
     @Override
     public Pair<Action, Direction> getActionWithDirection() {
-        return Pair.of(Action.MOVE, this.face);
+        if (this.face == Direction.UP || this.face == Direction.DOWN) {
+            return Pair.of(Action.IDLE, Direction.RIGHT);
+        }
+        return Pair.of(this.isMoving() ? Action.MOVE : Action.IDLE, this.face);
```

Comment:
Hehe this was me, here's the fix to make the boss not throw a NullPointerException on rendering